base/abstractarraymath.jl

```diff
@@ -407,6 +407,26 @@ _reperr(s, n, N) = throw(ArgumentError("number of " * s * " repetitions " *
     return R
 end
 
+struct EachSlice{A,I,L}
+    arr::A # underlying array
+    cartiter::I # CartesianIndices iterator
+    lookup::L # dimension look up: dimension index in cartiter, or nothing
+end
+
+function iterate(s::EachSlice, state...)
+    r = iterate(s.cartiter, state...)
+    isnothing(r) && return r
```

Comment:
We should probably use `=== nothing` everywhere instead of `isnothing` because of #27681. (Similarly applies to `nothing`/`isnothing`.)
```suggestion
    r === nothing && return r
```